cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -386,10 +384,10 @@ public static void main(String[] args) throws Exception {
    * Sets up a {@link DatasetFramework} instance for standalone usage.  NOTE: should NOT be used by applications!!!
    */
   private DatasetFramework createRegisteredDatasetFramework(CConfiguration cConf,
-                                                            DatasetDefinitionRegistryFactory registryFactory)
+                                                            DatasetDefinitionRegistryFactory registryFactory,
+                                                            Map<String, DatasetModule> defaultModules)
     throws DatasetManagementException, IOException {
-    DatasetFramework datasetFramework = new InMemoryDatasetFramework(registryFactory, cConf);
-    addModules(datasetFramework);
+    DatasetFramework datasetFramework = new InMemoryDatasetFramework(registryFactory, defaultModules, cConf);
     // dataset service
     DatasetMetaTableUtil.setupDatasets(datasetFramework);
```

Comment:
Actually we can remove all these together since the defaultModules injection is done through a multibinder. Anyway, we can do it separate from this PR.